components/filecoin/rs/src/signature.rs

```diff
@@ -0,0 +1,187 @@
+use crate::message::MessageAPI;
+use blake2b_simd::Params;
+use bls_signatures::Serialize;
+use core::{array::TryFromSliceError, num::ParseIntError};
+use fvm_ipld_encoding::Cbor;
```

Comment:
How hard would it be for us to eliminate the fvm dependencies here? This [brings in quite a few transient dependencies](https://github.com/brave/brave-core/blob/0b87e3e0c37276742aa78f84ffe22cd8269bd789/build/rust/Cargo.lock#L991-L1048) from the looks of it

Reply:
if you mean all fvm_* then this will be hard
```
use fvm_ipld_encoding::Cbor;
use fvm_shared::address::Protocol;
use fvm_shared::message::Message as UnsignedMessage;
```

Reply:
Yeah that's what I was hoping for but I hadn't looked into what was happening under the hood for those APIs. No worries we can review it instead